codegenerator/cli/templates/static/codegen/src/eventFetching/ChainFetcher.res

```diff
@@ -283,6 +283,12 @@ let hasProcessedToEndblock = (self: t) => {
   }
 }
 
+let hasNoMoreEventsToProcess = (self: t, ~hasArbQueueEvents) => {
+  !hasArbQueueEvents &&
+  !self.isFetchingBatch &&
```

Comment:
Added the line

Reply:
Ok cool so the issue was that it was hitting a false positive since it was still busy fetching a batch?